ph5/clients/ph5tostationxml.py

```diff
@@ -658,69 +658,76 @@ def read_stations(self):
-                        station_entry = station_list[deployment][0]
-                        sta_longitude = station_entry['location/X/value_d']
-                        sta_latitude = station_entry['location/Y/value_d']
-                        sta_elevation = station_entry['location/Z/value_d']
-
-                        if not self.is_lat_lon_match(sta_xml_obj,
-                                                     sta_latitude,
-                                                     sta_longitude):
-                            continue
-
-                        if station_entry['seed_station_name_s']:
-                            station_code = station_entry['seed_station_name_s']
-                        else:
-                            station_code = sta_id
-
-                        start_date = station_entry['deploy_time/epoch_l']
-                        start_date = UTCDateTime(start_date)
-                        end_date = station_entry['pickup_time/epoch_l']
-                        end_date = UTCDateTime(end_date)
-                        if sta_xml_obj.start_time and \
-                                sta_xml_obj.start_time > end_date:
-                            # chosen start time after pickup
-                            continue
-                        elif sta_xml_obj.end_time and \
-                                sta_xml_obj.end_time < start_date:
-                            # chosen end time before pickup
-                            continue
-                        creation_date = start_date
-                        termination_date = end_date
-                        site_name = station_entry['location/description_s']
-
-                        obs_channels = []
-                        # run channel filters if necessary. we do this first
-                        # to avoid creating a station that has no channels
-                        if self.manager.level.upper() == "RESPONSE" or \
-                           self.manager.level.upper() == "CHANNEL" or \
-                           sta_xml_obj.location_list != ['*'] or \
-                           sta_xml_obj.channel_list != ['*'] or \
-                           sta_xml_obj.component_list != ['*'] or \
-                           sta_xml_obj.receiver_list != ['*']:
-                            obs_channels = self.read_channels(sta_xml_obj,
-                                                              station_list,
-                                                              deployment,
-                                                              station_code,
-                                                              array_code)
-                            # go to the next station if no channels were
-                            # returned
-                            if len(obs_channels) == 0:
+                        station_epochs = station_list[deployment]
+                        for station_entry in station_epochs:
```

Comment:
This function is too long and nested. Refactoring here may not be the place but you can reduce the nest (and all the line continuations) with
```
for deployment, station_epoch, station entry in ( (dk, dv, se) for dk, dv in station_list.items() for se in dv )
```